include/deal.II/grid/reference_cell.h

```diff
@@ -919,6 +1032,323 @@ ReferenceCell::standard_line_to_face_and_line_index(
+        face,
+        line,
+        Utilities::get_bit(face_orientation, 0),
+        Utilities::get_bit(face_orientation, 2),
+        Utilities::get_bit(face_orientation, 1));
+    }
+  else if (*this == ReferenceCells::Tetrahedron)
+    {
+      const static unsigned int table[4][3] = {{0, 1, 2},
+                                               {0, 3, 4},
+                                               {2, 5, 3},
+                                               {1, 4, 5}};
+
+      return table[face]
+                  [standard_to_real_face_line(line, face, face_orientation)];
+    }
+  else if (*this == ReferenceCells::Pyramid)
+    {
+      Assert(false, ExcNotImplemented());
+    }
+  else if (*this == ReferenceCells::Wedge)
+    {
+      Assert(false, ExcNotImplemented());
+    }
+  else if (*this == ReferenceCells::Hexahedron)
+    {
+      return GeometryInfo<3>::face_to_cell_lines(
+        face,
+        line,
+        Utilities::get_bit(face_orientation, 0),
+        Utilities::get_bit(face_orientation, 2),
+        Utilities::get_bit(face_orientation, 1));
+    }
+
+  Assert(false, ExcNotImplemented());
+  return numbers::invalid_unsigned_int;
+}
+
+
+
+inline unsigned int
+ReferenceCell::face_to_cell_vertices(const unsigned int  face,
+                                     const unsigned int  vertex,
+                                     const unsigned char face_orientation) const
+{
+  AssertIndexRange(face, n_faces());
+  AssertIndexRange(vertex, face_reference_cell(face).n_vertices());
+
+  if (*this == ReferenceCells::Vertex)
+    {
+      Assert(false, ExcNotImplemented());
+    }
+  else if (*this == ReferenceCells::Line)
+    {
+      return GeometryInfo<1>::face_to_cell_vertices(
+        face,
+        vertex,
+        Utilities::get_bit(face_orientation, 0),
+        Utilities::get_bit(face_orientation, 2),
+        Utilities::get_bit(face_orientation, 1));
+    }
+  else if (*this == ReferenceCells::Triangle)
+    {
+      static const unsigned int table[3][2] = {{0, 1}, {1, 2}, {2, 0}};
+
+      return table[face][face_orientation ? vertex : (1 - vertex)];
+    }
+  else if (*this == ReferenceCells::Quadrilateral)
+    {
+      return GeometryInfo<2>::face_to_cell_vertices(
+        face,
+        vertex,
+        Utilities::get_bit(face_orientation, 0),
+        Utilities::get_bit(face_orientation, 2),
+        Utilities::get_bit(face_orientation, 1));
+    }
+  else if (*this == ReferenceCells::Tetrahedron)
+    {
+      static const unsigned int table[4][3] = {{0, 1, 2},
+                                               {1, 0, 3},
+                                               {0, 2, 3},
+                                               {2, 1, 3}};
+
+      return table[face][standard_to_real_face_vertex(
+        vertex, face, face_orientation)];
+    }
+  else if (*this == ReferenceCells::Pyramid)
+    {
+      constexpr auto            X           = numbers::invalid_unsigned_int;
```

Comment:
We could move the x to the top of the function.

Reply:
I think it's fine as is -- it makes it easy to read what the table says without having to scroll to the top of the page. (I actually kind of like your approach with the `X` as an invalid placeholder in tables.)